libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigationProvider.kt

```diff
@@ -0,0 +1,57 @@
+package com.mapbox.navigation.core
+
+import android.content.Context
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEngineProvider
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.navigation.base.options.NavigationOptions
+
+
+object MapboxNavigationProvider {
+    @Volatile
+    private var mapboxNavigation: MapboxNavigation? = null
+
+    /**
+     *  Pass NavigationInitializationBundle to initialize a new instance of MapboxNavigation
+     *  Old instance will be cleaned up, if exists.
+     *  Don't pass any params to get current MapboxNavigation instance
+     */
+    fun navigation(bundle: NavigationInitializationBundle? = null): MapboxNavigation {
```

Comment:
nit: it's not going to be clear what is responsible for the "initial class load". As in, what should create the initial option bundle? with separate functions we can provide better recommendations

``` kotlin
object MapboxNavigationProvider {
   /** recommended to call in Application.onCreate **/
   fun createNavigation(options: Options): MapboxNavigation
   /** can call anywhere, won't work unless createNavigation is called **/
   fun navigation(): MapboxNavigation
}
```

Reply:
done